docroot/modules/custom/sitenow_signage/sitenow_signage.module

```diff
@@ -338,34 +323,124 @@ function sitenow_signage_preprocess_page(&$variables) {
 function sitenow_signage_preprocess_paragraph(&$variables) {
   $paragraph = $variables['paragraph'];
 
-  if ($paragraph->bundle() == 'slide_image' &&
-    isset($variables['content']['field_slide_image'])) {
+  switch ($paragraph->bundle()) {
+    case 'slide_image':
+      if (isset($variables['content']['field_slide_image'])) {
 
-    // Get the sign node from the current route.
-    $sign_node = \Drupal::routeMatch()->getParameter('node');
+        // Get the sign node from the current route.
+        $sign_node = \Drupal::routeMatch()->getParameter('node');
 
-    if ($sign_node && $sign_node->bundle() == 'sign') {
-      $variables['#cache']['tags'][] = 'node:' . $sign_node->id();
-    }
+        if ($sign_node && $sign_node->bundle() == 'sign') {
+          $variables['#cache']['tags'][] = 'node:' . $sign_node->id();
+        }
+
+        // Check if we have a sign node with vertical orientation.
+        if ($sign_node &&
+          $sign_node->bundle() == 'sign' &&
+          $sign_node->hasField('field_sign_orientation') &&
+          !$sign_node->get('field_sign_orientation')->isEmpty() &&
+          $sign_node->get('field_sign_orientation')->value === 'vertical') {
```

Comment:
Seems like this could use the `sitenow_signage_get_orientation` helper function here.

Reply:
Also for what its worth, we typically write this type of accessor as `$sign_node->field_sign_orientation->value`.

Reply:
That helper function only takes a paragraph, and is more intended to traverse up and get the orientation of a sign that contains the paragraph that we pass as an argument.

Perhaps it should be named something like, `sitenow_signage_get_paragraph_sign_orientation()`?